multichain-testing/test/tools/wallet.test.ts

```diff
@@ -0,0 +1,41 @@
+import anyTest from '@endo/ses-ava/prepare-endo.js';
+import type { TestFn } from 'ava';
+import { makeQueryClient } from '../../tools/query.js';
+import { createWallet } from '../../tools/wallet.js';
+import { sleep } from '../../tools/sleep.js';
+import { commonSetup } from '../support.js';
+
+const test = anyTest as TestFn<Record<string, never>>;
+
+test('create a wallet and get tokens', async (t) => {
+  const { useChain } = await commonSetup(t);
+
+  const prefix = useChain('osmosis').chain.bech32_prefix;
+  const wallet = await createWallet(prefix);
+  const addr = (await wallet.getAccounts())[0].address;
+  t.regex(addr, /^osmo1/);
+  t.log('Made temp wallet:', addr);
+
+  const apiUrl = useChain('osmosis').getRestEndpoint();
+  const queryClient = makeQueryClient(apiUrl);
+  t.log('Made query client');
+
+  const { balances } = await queryClient.queryBalances(addr);
+  t.log('Beginning balances:', balances);
+  t.deepEqual(balances, []);
+
+  const osmosisFaucet = useChain('osmosis').creditFromFaucet;
+  t.log('Requeting faucet funds');
+  await osmosisFaucet(addr);
+  await sleep(1000, t.log); // needed to avoid race condition
```

Comment:
between what? is there something we can await?

Reply:
The `osmosisFaucet` call is a POST request to the faucet server, and the delay is needed for the API query. It's unclear if the delay is transaction inclusion or something else.

Agree this would be nice to remove, or abstract away in test setup.

Reply:
please document with an `XXX` to help someone debugging the race or who wants to clean it up